libmamba/src/core/package_handling.cpp

```diff
@@ -354,51 +437,74 @@ namespace mamba
         LOG_INFO << "Extracting " << file << " to " << destination;
         extraction_guard g(destination);
 
+        int r;
+
+        libarchive_read_raii a;
+        archive_read_support_format_tar(a);
+        archive_read_support_format_zip(a);
+        archive_read_support_filter_all(a);
+
+        auto lock = LockFile(file);
+        r = archive_read_open_filename(a, file.string().c_str(), 10240);
+
+        if (r != ARCHIVE_OK)
+        {
+            LOG_ERROR << "Error opening archive: " << archive_error_string(a);
+            throw std::runtime_error(file.string() + " : Could not open archive for reading.");
+        }
+
+        stream_extract_archive(a, destination);
+    }
+
+    namespace
+    {
+        struct conda_extract_context
```

Comment:
Consider making this type non-copyable explicitly (`= delete;`)

Reply:
done